lib/matrix_convert_to_zero.rb

```diff
@@ -3,8 +3,34 @@
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^3) - n * n * m, where n is the length of the largest row/col
+# and m is the length of the smallest row/col
+# Space complexity: O(1) - constant
 def matrix_convert_to_zero(matrix)
-  raise NotImplementedError
+  rows = matrix.length
+  cols = matrix[0].length
+
+  rows.times do |row|
+    cols.times do |col|
+      matrix[row][col] = "x" if matrix[row][col] == 0
```

Comment:
Good use of temporary state tracking here! 👍

(This technique becomes super-relevant when you start working with things like graphs and trees.)